src/test/acceptance/features/navigation.feature

```diff
@@ -0,0 +1,52 @@
+Feature: Application process navigation is controlled
+  In order to facilitate the beta HTBHF programme
+  As a potential claimant
+  I want to make sure that only the correct application flow can be used
+
+  Scenario Outline: Enter name page is always the first page
+    Given I am starting a new application
+    When I navigate to the <page> page
+    Then I am shown the enter name page
+    Examples:
+      | page                     |
+      | enter name               |
+      | enter national insurance |
+      | enter date of birth      |
+      | are you pregnant         |
+      | card address             |
+
+  Scenario Outline: Navigation after completing application returns to confirm page
+    Given I have completed my application
+    When I navigate to the <page> page
+    Then I am shown a successful confirmation page
+    Examples:
+      | page                     |
+      | enter name               |
+      | enter national insurance |
+      | enter date of birth      |
+      | are you pregnant         |
+      | card address             |
+
+  Scenario Outline: Navigation is not allowed past the current page in the flow
+    Given I have entered my details up to the <page> page
+    When I navigate to the check details page
+    Then I am shown the <page> page
+    Examples:
+      | page                     |
+      | enter name               |
+      | enter national insurance |
+      | enter date of birth      |
+      | are you pregnant         |
+      | card address             |
+
+  Scenario Outline: Navigation to previous steps in an unsubmitted application flow is allowed
+    Given I have entered my details up to the check details page
+    When I navigate to the <page> page
+    Then I am shown the <page> page
```

Comment:
This might be quite a lengthy test. Wish there was a way to loop over steps within a scenario - so that the given was executed once and the when/then once for each example. Short of that I guess we just keep an eye on how long the tests take.

Reply:
Yes, I think I'm more of a mind to keep the tests simple first, and if they get too long then refactor later. On my laptop they seemed to be quite speedy, but I guess it has more hardware than the Travis build servers.